### PR TITLE
Prettier tests

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- Add testing resource crd and config map
- Add tests for testing resource with important functionality tested:
  - dynamicValue
  - Variables
  - defaultValue
  - presets (in pizzas we have presets so we can change them and check if values are correct)
  - templates (to GenericList in Create)
  - data sources

**Related issue(s)**

### CODE REVIEW

File: tests/integration/tests/extensibility/ext-test-variables.spec.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+    // test vars with defaultValue
+    cy.get('@form')
+      .find('[data-testid="varWithDefaultValue"]:visible')
+      .should('have.value', 'default');
+
+    // test vars with dynamicValue
+    cy.get('@form')
+      .find('[data-testid="varWithDynamicValue"]:visible')
+      .should('have.value', 'default name');
```
Comment (Lyczeq) [inline tests/integration/tests/extensibility/ext-test-variables.spec.js:145]: I'd propose to change the value of the dynamic input from `default name` to `dynamic name`.

File: tests/integration/tests/extensibility/ext-test-variables.spec.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    // test default preset
+    cy.get('@form')
+      .find('[arialabel="TestResource name"]:visible')
+      .should('have.value', NAME);
+
+    // test presets
+    cy.get('@form')
+      .find('[data-testid="presets"]:visible')
+      .find('.fd-button.fd-select__button')
+      // .find('[aria-label="Combobox input arrow"]:visible', { log: false })
```
Comment (Lyczeq) [inline tests/integration/tests/extensibility/ext-test-variables.spec.js:159]: If the comment is not necessary, please remove it.

File: tests/integration/tests/extensibility/ext-test-variables.spec.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+      .click();
+
+    cy.get('[role="list"]')
+      .contains('Fixes')
+      .should('exist');
+
+    cy.get('[role="list"]')
+      .contains('Fixes')
+      .click();
```
Comment (Lyczeq) [inline tests/integration/tests/extensibility/ext-test-variables.spec.js:168]: I think there's no need to check whether element exists if you want it to be clicked later.

File: tests/integration/tests/extensibility/ext-test-variables.spec.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+    cy.get('@form')
+      .find('[data-testid="spec.prefix"]:visible')
+      .should('have.value', 'prefix');
+
+    cy.get('@form')
+      .find('[data-testid="spec.suffix"]:visible')
+      .should('have.value', 'suffix');
+  });
```
Comment (Lyczeq) [inline tests/integration/tests/extensibility/ext-test-variables.spec.js:177]: I see that `Fixes` template clears the name of the resource. Could you test dynamicValue input to have `unnamed` value? I think it's a good use case.

**Review by Lyczeq — CHANGES_REQUESTED** [review]

File: tests/integration/tests/extensibility/ext-test-variables.spec.js [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+    cy.getLeftNav()
+      .contains('Test Resources')
+      .click();
+
+    cy.contains('Create Test Resource').click();
+
+    cy.get('.fd-dialog__content').as('form');
+  });
+
```
[the remark this answers is not in the document]
Comment (akucharska) [reply tests/integration/tests/extensibility/ext-test-variables.spec.js:79]: This way we know the navigation node is created properly. And we can not use `cy.goToNamespaceDetails` cause it uses `Cypress.env('NAMESPACE_NAME')`

[the remark this answers is not in the document]
Comment (Lyczeq) [reply tests/integration/tests/extensibility/ext-test-variables.spec.js]: `cy.navigateTo` actually will work, but it's up to you if you want to use it. It's already created and will take less code. Fair point about `cy.goToNamespaceDetails`, forgot about it.

File: tests/integration/tests/extensibility/ext-test-variables.spec.js [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+    cy.get('@form')
+      .find('[arialabel="TestResource name"]:visible')
+      .should('have.value', NAME);
+
+    // test presets
+    cy.get('@form')
+      .find('.fd-select__text-content:visible')
+      .contains('Choose preset')
+      .next('.fd-button.fd-select__button')
+      .click();
```
Comment (Lyczeq) [inline tests/integration/tests/extensibility/ext-test-variables.spec.js:156]:
```suggestion
    cy.get('@form')
      .contains('Choose preset')
      .click();
```

I think it's enough to find the preset dropdown and click it.

**Review by Lyczeq — APPROVED** [review]

**Pull request merged**